teachopencadd/talktorials/T000_template/talktorial.ipynb

```diff
@@ -179,15 +362,63 @@
    "metadata": {},
    "outputs": [],
    "source": [
-    "# Add comments in the Code cell if you want to comment on coding decisions"
+    "# Add comments in the code cell if you want to comment on coding decisions"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "<div class=\"alert alert-block alert-info\">\n",
+    "\n",
+    "<b>Functions</b>: \n",
+    "\n",
+    "<ul>\n",
+    "<li>Please add <a href=\"https://numpydoc.readthedocs.io/en/latest/format.html\">numpy docstrings</a> to your functions.</li>\n",
+    "<li>Please expose all variables used within a function in the function's signature (i.e. they must be function parameters), unless they are created within the scope of the function.</li>\n",
+    "<li>Please add comments to the steps performed in the function.</li>\n",
+    "</ul>\n",
+    "    \n",
+    "</div>"
    ]
   },
   {
    "cell_type": "code",
-   "execution_count": null,
+   "execution_count": 4,
    "metadata": {},
    "outputs": [],
-   "source": []
+   "source": [
+    "def calculate_ro5_properties(smiles):\n",
+    "    \"\"\"\n",
+    "    Test if input molecule (SMILES) fulfills Lipinski's rule of five.\n",
+    "\n",
+    "    Parameters\n",
+    "    ----------\n",
+    "    smiles : str\n",
+    "        SMILES for a molecule.\n",
+    "\n",
+    "    Returns\n",
+    "    -------\n",
+    "    pandas.Series\n",
+    "        Molecular weight, number of hydrogen bond acceptors/donor and logP value\n",
+    "        and Lipinski's rule of five compliance for input molecule.\n",
+    "    \"\"\"\n",
+    "    # RDKit molecule from SMILES\n",
+    "    molecule = Chem.MolFromSmiles(smiles)\n",
+    "    # Calculate Ro5-relevant chemical properties\n",
+    "    molecular_weight = Descriptors.ExactMolWt(molecule)\n",
+    "    n_hba = Descriptors.NumHAcceptors(molecule)\n",
+    "    n_hbd = Descriptors.NumHDonors(molecule)\n",
+    "    logp = Descriptors.MolLogP(molecule)\n",
+    "    # Ro5 conditions fulfilled?\n",
+    "    conditions = [molecular_weight <= 500, n_hba <= 10, n_hbd <= 5, logp <= 5]\n",
+    "    ro5_fulfilled = sum(conditions) >= 3\n",
+    "    # Return True if no more than one out of four conditions is violated\n",
+    "    return pd.Series(\n",
+    "        [molecular_weight, n_hba, n_hbd, logp, ro5_fulfilled],\n",
+    "        index=[\"molecular_weight\", \"n_hba\", \"n_hbd\", \"logp\", \"ro5_fulfilled\"],\n",
+    "    )"
+   ]
```

Comment:
Could we add before/after the example function, to use meaningful function and parameter names?